constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/utils/widget/ImageFilterView.java

```diff
@@ -399,6 +402,43 @@ public void setImageRotate(float rotation) {
         updateViewMatrix();
     }
 
+    @Override
```

Comment:
I wonder if it would be worth refactoring this into a small utility class you could use in both ImageFilterView and ImageFilterButton

Reply:
The code in theses two consist of 4  engines. 

1. ColorMatrix = 169LOC, (common)
2.  Rounding Corners = 172LOC, 
3. Transform = 27LOC, 
4.  Drawable IO= 37LOC

ColorMatrix is common,  calling it is 89LOC(v. 169) so it was worth it.
Transform call code is 90 (v. 27) Not really worth it.
Drawable IO is total 37 lines
Rounding corners is a series of calls to super class API I am not sure if it would work.

The bulk of the code is the access API and documentation.
If we had multiple inheritance it would scale.
So the balance of what is the value of reuse. Would it pay to reuse the 27 lines of Transform?
It did not reach my threshold.  Rounding it is reaching that point. 
You could bundle all of this into a utility object  (ImageFilterSupport) saving about 200 of the 700 lines. 
Maybe that is a good exercise for another day.